### PR TITLE
Add ability to read LMDB without any labels stored inside

### PR DESCRIPTION
**Pull request by JanuszL**
#### Why we need this PR?
 - now DALI assumes that labels and images are stored together in the LMDB for Caffe and Caffe2 formats. This is not always true
 - adds an ability to specify if images and labels are really inside LMDB

#### What happened in this PR?
 - now DALI assumes that labels and images are stored together in the LMDB for Caffe and Caffe2 formats. This is not always true. This PR adds an ability to specify if images and labels are really inside LMDB
 - Tested in CI, https://github.com/NVIDIA/DALI_extra/pull/30 needs to be approved as well
 - Ad https://github.com/NVIDIA/DALI/issues/1433

**JIRA TASK**: [DALI-1135]

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/caffe2_reader_op.cc [hunk 1]
```diff
@@ -23,13 +23,15 @@ DALI_SCHEMA(Caffe2Reader)
...
   .NumInput(0)
   .OutputFn([](const OpSpec& spec) {
+      int img_idx = spec.GetArgument<bool>("image_available") ? 1 : 0;
       auto label_type = static_cast<LabelType>(spec.GetArgument<int>("label_type"));
 
-      int num_label_outputs = (label_type == MULTI_LABEL_SPARSE ||
-                               label_type == MULTI_LABEL_WEIGHTED_SPARSE) ? 2 : 1;
+      int num_label_outputs = (label_type == NO_LABEL) ? 0 : 1;
+      num_label_outputs += (label_type == MULTI_LABEL_SPARSE ||
+                               label_type == MULTI_LABEL_WEIGHTED_SPARSE) ? 1 : 0;
```
Comment (jantonguirao) [inline dali/operators/reader/caffe2_reader_op.cc]: nitpick: indentation is off

File: dali/test/python/test_pipeline.py [hunk 2]
```diff
@@ -1395,6 +1397,62 @@ def test_skip_cached_images():
...
+
+        def define_graph(self):
+            if not self.labels:
+                jpegs = self.input()
+            else:
+                jpegs,_ = self.input()
+            images = self.decode(jpegs)
+            return (images)
+
+    pipe = CaffePipeline(2, caffe_db_folder, True)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]: you could use compare_pipelines to compare that the outputs are the same

File: dali/test/python/test_pipeline.py [hunk 3]
```diff
@@ -1395,6 +1397,62 @@ def test_skip_cached_images():
...
+
+        def define_graph(self):
+            if self.label_type == 4:
+                jpegs = self.input()
+            else:
+                jpegs,_ = self.input()
+            images = self.decode(jpegs)
+            return (images)
+
+    pipe = Caffe2Pipeline(2, c2lmdb_db_folder, 0)
```
Comment (jantonguirao) [inline dali/test/python/test_pipeline.py]: and here as well

File: qa/setup_dali_extra.sh [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 # Fetch test data
 export DALI_EXTRA_PATH=${DALI_EXTRA_PATH:-/opt/dali_extra}
-export DALI_EXTRA_URL=${DALI_EXTRA_URL:-"https://github.com/NVIDIA/DALI_extra.git"}
+export DALI_EXTRA_URL=https://github.com/JanuszL/DALI_extra/
```
Comment (jantonguirao) [inline qa/setup_dali_extra.sh]: revert this file

File: dali/operators/reader/caffe2_reader_op.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/caffe2_reader_op.cc]: Done

File: dali/test/python/test_pipeline.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/test_pipeline.py]: Not really, no_label data set is a small subset of the bigger one. I though there is no point in putting all images there just to check if it works without labels.

File: dali/test/python/test_pipeline.py (hunk 3 above)
Comment (JanuszL) [reply dali/test/python/test_pipeline.py]: As above

File: qa/setup_dali_extra.sh (hunk 4 above)
Comment (JanuszL) [reply qa/setup_dali_extra.sh]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**